hs_core/models.py

```diff
@@ -1813,11 +1817,22 @@ def getAVU(self, attribute):
         istorage = self.get_irods_storage()
         root_path = self.root_path
         value = istorage.getAVU(root_path, attribute)
+
+        # Convert selected boolean attribute values to bool; non-existence implies False
         if attribute == 'isPublic':
-            if value.lower() == 'true':
+            if value is not None and value.lower() == 'true':
                 return True
             else:
                 return False
+
+        # Convert selected boolean attribute values to bool; non-existence implies True
+        if attribute == 'bag_modified' or attribute == 'metadata_dirty':
+            if value is None or value.lower() == 'true':
+                return True
```

Comment:
Nope. I made this mistake the first time. I am correcting it in this version. 

To avoid botches, I interpret the non-existence of a flag as 'true' if the flag is an action flag, and 'false' if the flag is a state flag. Thus, my code automatically makes the assumption that any non-existent flag has the most conservative value, and recovers instantly from any lack of flags.

It was previously what you suggest, so that non-existent flags weren't interpreted properly. If bag_modified is empty, the default should be to generate the bag, on the logic that we don't know the state and should recover to a situation in which we know state. 

Perhaps I should explain with a comment.